app/scripts/modules/google/src/loadBalancer/configure/internal/listeners/listener.component.html

```diff
@@ -0,0 +1,97 @@
+          <strong>Your listener will be named:</strong>
+          <span>{{$ctrl.getName($ctrl.listener, $ctrl.application.name)}}</span>
+          <!-- Angular does not seem to run length validation on hidden inputs, hence the text + display:none -->
+          <input type="text"
+                 style="display: none"
+                 ng-maxlength="63"
+                 class="form-control input-sm"
+                 ng-model="$ctrl.listener.name"
+                 ng-readonly="$ctrl.listener.created"
+                 validate-unique="$ctrl.existingListenerNames()"
+                 validate-ignore-case="true"
+                 name="listenerName"/>
+          <validation-error ng-if="listener.listenerName.$error.validateUnique || $ctrl.localListenerHasSameName()"
+                            message="There is already a listener in {{$ctrl.command.loadBalancer.credentials}} with that name.">
+          </validation-error>
+        </div>
+
+        <div class="col-md-2 sm-label-right">Stack <help-field key="aws.loadBalancer.stack"></help-field></div>
+        <div class="col-md-3">
+          <input type="text"
+                 class="form-control input-sm"
+                 ng-model="$ctrl.listener.stack"
+                 ng-readonly="$ctrl.listener.created"
+                 name="stackName"
+                 ng-change="$ctrl.updateName($ctrl.listener, $ctrl.application.name)"
+                 ng-pattern="/^[a-z0-9]*$/"/>
+        </div>
+        <div class="col-md-2 sm-label-right">Detail <help-field key="gce.loadBalancer.detail"></help-field></div>
+
+        <div class="col-md-3">
+          <input type="text"
+                 class="form-control input-sm"
+                 ng-model="$ctrl.listener.detail"
+                 ng-readonly="$ctrl.listener.created"
+                 name="detailName"
+                 ng-change="$ctrl.updateName($ctrl.listener, $ctrl.application.name)"
+                 ng-pattern="/^[a-z0-9-]*$/"/>
+        </div>
+
+        <div class="col-md-1">
+          <button class="btn btn-sm btn-default" ng-click="$ctrl.deleteListener()">
+            <span class="glyphicon glyphicon-trash visible-lg-inline"></span>
+            <span>Delete</span>
+          </button>
+        </div>
+
+        <table class="table table-condensed packed">
+          <thead>
+            <tr>
+              <th>Protocol</th>
+              <th>Subnet</th>
+              <th>Ports<help-field key="gce.internalLoadBalancer.ports"></help-field></th>
+            </tr>
+          </thead>
+          <tbody>
+          <tr>
+            <td>
+              <select ng-disabled="$ctrl.listener.created" class="form-control input-sm"
+                      ng-change="$ctrl.protocolUpdated()"
+                      ng-model="$ctrl.loadBalancer.ipProtocol"
+                      ng-options="protocol for protocol in ['TCP', 'UDP']"></select>
```

Comment:
Is this a per-listener setting? Or is it set for the load balancer? (It's being set on the load balancer itself right now).